src/niscope/system_tests/test_system_niscope.py

```diff
@@ -1,527 +1,588 @@
+
+        for sample in waveform:
+            assert not math.isnan(sample)
+        assert len(waveforms) == test_num_channels * test_num_records_to_fetch
+
+        expected_channels = self._test_channels.split(',') * test_num_records_to_fetch
+        expected_records = [2, 2, 3, 3, 4, 4]
+        for i in range(len(waveforms)):
+            record_wfm = waveforms[i].samples
+            assert len(record_wfm) == test_record_length
+            for j in range(len(record_wfm)):
+                assert record_wfm[j] == waveform[i * test_record_length + j]
+            assert waveforms[i].channel == expected_channels[i]
+            assert waveforms[i].record == expected_records[i]
+
+    def test_error_message(self):
+        try:
+            # We pass in an invalid model name to force going to error_message
+            with niscope.Session('FakeDevice', False, True, 'Simulate=1, DriverSetup=Model:invalid_model; BoardType:PXIe'):
+                assert False
+        except niscope.Error as e:
+            assert e.code == -1074118609
+            assert e.description.find('Simulation does not support the selected model and board type.') != -1
+
+    def test_get_self_cal_last_date_time(self, single_instrument_session):
+        last_cal = single_instrument_session.get_self_cal_last_date_and_time()
+        assert last_cal.month == 12
+        assert last_cal.day == 21
+        assert last_cal.year == 1999
+        assert last_cal.hour == 0
+        assert last_cal.minute == 0
+
+    def test_get_ext_cal_last_date_time(self, single_instrument_session):
+        last_cal = single_instrument_session.get_ext_cal_last_date_and_time()
+        assert last_cal.month == 12
+        assert last_cal.day == 21
+        assert last_cal.year == 1999
+        assert last_cal.hour == 0
+        assert last_cal.minute == 0
+
+    def test_get_self_cal_last_temperature(self, single_instrument_session):
+        last_cal_temp = single_instrument_session.get_self_cal_last_temp()
+        assert last_cal_temp == 25
+
+    def test_get_ext_cal_last_temperature(self, single_instrument_session):
+        last_cal_temp = single_instrument_session.get_ext_cal_last_temp()
+        assert last_cal_temp == 25
+
+    def test_configure_ref_levels(self, single_instrument_session):
+        single_instrument_session._configure_ref_levels()
+        assert 90.0 == single_instrument_session.meas_chan_high_ref_level
+
+    @pytest.fixture(params=[(1000, 1000), (2000, 2000), (3000, 2000)], ids=["less_than_actual", "equal_to_actual", "greater_than_actual"])
+    def measurement_wfm_length(self, request):
+        MeasWfmLength = collections.namedtuple('MeasurementWaveformLength', ['passed_in', 'expected'])
+        return MeasWfmLength(passed_in=request.param[0], expected=request.param[1])
+
+    # TODO(RyanEckenrode): AB#2207665: Need to investigate how to get FetchArrayMeasurement working for the grpc versions of these tests
```

Comment:
Note: Added a TODO here and the task number. Basically, there's a difference in the FetchArrayMeasurement method params between nimi-python and grpc-device. In grpc-device, currently we don't have the `measurement_wfm_length` as an input in the .proto file. We just calculate it server-side by calling `ActualMeasWfmSize`. But on nimi-python, it accepts the `measurement_wfm_size` from the client.

Made the TODO task to investigate this and figure out what should be done. Part of that Task is to enable these two `fetch_array_measurement` tests for gRPC as well.

Reply:
These tests are not going to be enabled for grpc but there is still work to be done to have better behavior when a grpc session is used so I have left the todo in for now

Reply:
@marcoskirsch has requested that we error when passing measurement_wfm_length while using gRPC. Since the TODO work item is parented under the overall work for adding gRPC NI-SCOPE support, though, I'll consider this resolved for the purpose of this PR.